File: gen.py
import random
def isSpecial(char):
    ranges = [48, 57, 65, 90, 97, 122]
    # numbers between each set of 2 are non special characters
    for i in range(len(ranges)):
        if i % 2 == 0:
            if char >= ranges[i] and char <= ranges[i + 1]:
                return False
    return True
def newCharacter():
    return random.random() * 94 + 33
def genString(length, special):
	password = ""
	for i in range(length):
		newChar = newCharacter()
		if not special:
			while isSpecial(newChar):
				newChar = newCharacter()
		password += chr(int(newChar))
	return password
if __name__ == "__main__":
	while True:
		length = input("Length: ")
		try:
			length = int(length)
			print(genString(length, True)) # special characters
			print(genString(length, False)) # no special characters
			print("")
		except:
			print("")

File: gen.js
var length = 32
function isSpecial(char) {
	var ranges = [48, 57, 65, 90, 97, 122]
	// numbers between each set of 2 are non special characters
	for (var i = 0; i < ranges.length; i ++) {
		if (i % 2 == 0) {
			if (char >= ranges[i] && char <= ranges[i + 1]) {
				return false
			}
		}
	}
	return true
}

newCharacter = () => Math.floor(Math.random() * 94 + 33)

function genString(length, special) {
	password = ""
	for (var i = 0; i < length; i++) {
		var newChar = newCharacter()
		if (!special) {
			while (isSpecial(newChar)) {
				newChar = newCharacter()
			}
		}
		password += String.fromCharCode(newChar)
	}
	return password
}

console.log(genString(length, true)) // special characters
console.log(genString(length)) // no special characters
